src/Asset/Widget/Queue.php

```diff
@@ -156,23 +156,23 @@ public function countItemsInQueue($location, $type = 'frontend')
      *
      * @return string|null
      */
-    public function render($location, $type = 'frontend')
+    public function render($location, $type = 'frontend', $wrappertemplate = '', $holdertemplate = '')
     {
         $html = null;
 
         /** @var WidgetAssetInterface $widget */
         foreach ($this->sort($this->queue) as $widget) {
             if ($widget->getType() === $type && $widget->getLocation() === $location) {
-                $html .= $this->addWidgetHolder($widget);
+                $html .= $this->addWidgetHolder($widget, $holdertemplate);
             }
         }
 
         if ($html !== null) {
-            $html = sprintf(
-                '<div class="widgetholder widgetholder-%s">%s</div>',
-                $location,
-                $html
-            );
+            if (empty($wrappertemplate)) {
+                $wrappertemplate = 'widgetholder.twig';
```

Comment:
On that last one, yes… but `$holderTemplate` should be required or default to `null`, and certainly be required in the protected function.